app/src/main/java/org/stepic/droid/storage/StoreStateManagerImpl.java

```diff
@@ -53,18 +56,25 @@ public void updateUnitLessonState(long lessonId) {
             analytic.reportError(Analytic.Error.LESSON_IN_STORE_STATE_NULL, new NullPointerException("lesson was null"));
             return;
         }
-        lesson.set_loading(false);
-        lesson.set_cached(true);
+        lesson.set_loading(loading);
+        lesson.set_cached(cached);
```

Comment:
//Please add this to comment
cached = true -> loading = false
cached = false -> loading = false|true